src/jit/optimizer.cpp

```diff
@@ -8910,15 +8910,16 @@ void Compiler::optMergeBoolReturns()
                 continue;
             }
 
-            if (!returnOp->gtGetOp1()->OperIsCompare() || varTypeIsFloating(returnOp->gtGetOp1()->TypeGet()))
+            if (!(returnOp->gtGetOp1()->OperIsCompare() || returnOp->gtGetOp1()->OperIs(GT_NEG)) ||
+                varTypeIsFloating(returnOp->gtGetOp1()->TypeGet()))
             {
                 continue;
             }
 
             if (!jumpBbReturnValue)
             {
                 // if jump target bb is `return false` we need to inverse comparison for our future `return expr`
-                if (!returnOp->gtGetOp1()->OperIs(GT_EQ, GT_NE, GT_GT, GT_GE, GT_LT, GT_LE, GT_TEST_EQ, GT_TEST_NE))
+                if (!returnOp->gtGetOp1()->OperIs(GT_EQ, GT_NE))
```

Comment:
Why do you use
`OperIs(GT_EQ, GT_NE, GT_GT, GT_GE, GT_LT, GT_LE, GT_TEST_EQ, GT_TEST_NE)`
above on line 8975 and now use
`OperIs(GT_EQ, GT_NE)`
here?

Reply:
@briansull for some reason the extended list here regresses the diffs (I didn't investigate why yet) but yeah `fgExpandBoolReturns` needs more work.